src/actions/attack.cpp

```diff
@@ -157,21 +157,27 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 	}
 
 	// Compute chance to hit.
-	chance_to_hit = opp.defense_modifier(resources::gameboard->map().get_terrain(opp_loc)) + weapon->accuracy()
+    int tmp_chance_to_hit = opp.defense_modifier(resources::gameboard->map().get_terrain(opp_loc)) + weapon->accuracy()
 		- (opp_weapon ? opp_weapon->parry() : 0);
 
-	if(chance_to_hit > 100) {
-		chance_to_hit = 100;
+	if(tmp_chance_to_hit > 100) {
+		tmp_chance_to_hit = 100;
 	}
 
 	unit_ability_list cth_specials = weapon->get_specials("chance_to_hit");
-	unit_abilities::effect cth_effects(cth_specials, chance_to_hit, backstab_pos);
-	chance_to_hit = cth_effects.get_composite_value();
+	unit_abilities::effect cth_effects(cth_specials, tmp_chance_to_hit, backstab_pos);
+	tmp_chance_to_hit = cth_effects.get_composite_value();
+		
+	if(tmp_chance_to_hit > 100) {
+		tmp_chance_to_hit = 100;
+	}
```

Comment:
It's a question of style but I would use utils::clamp here, too, rather than separate >100 and <0 checks.